### PR TITLE
.Net: Obsolete the useless SetDefaultFunctionCollection

### PR DESCRIPTION
**Pull request by stephentoub**
### Description

`ISKFunction` exposes a `SetDefaultFunctionCollection`. On all implementations it's either a nop or just delegates to the same interface method on a wrapped object. The one exception to that is `SemanticFunction`, which stores the argument into a field _that then is never used_.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by markwallace-microsoft — APPROVED** [review]

**Review by SergeyMenshykh — APPROVED** [review]

**Comment by dmytrostruk** [comment]
FYI: need re-approval after merge conflict resolution in this commit: https://github.com/microsoft/semantic-kernel/pull/3183/commits/39dbd018041593b5cbbe0ddb6cc7e163d9a1c77c

**Pull request merged**
